### ISSUE TITLE
[FP] Unnecessary type casting for `(int) ($foo[0] ?? 0)`

### ISSUE THREAD

**Issue by t-kanstantsin** [kalessil/phpinspectionsea#909]
Following code gives 'Unnecessary type casting' warning:
```php
(int) ($foo[0] ?? 0)
```
But it's true only for second part of statement.

PHPStorm version is 2018.1.2
Plugin version is 3.0.1

**Comment by funivan** [comment]
Thanks for report @t-kanstantsin 
Confirm this bug.

**Comment by kalessil** [comment]
Will check, thank you for reporting.

**Comment by foobar13373** [comment]
Can confirm, was just about to report the same issue, only for `(bool)` instead of `(int)`.

**Comment by kalessil** [comment]
For myself: when about to report, run possible values discovery and skip reporting for ?? (PhpStorm types resolvers is not yet efficient for ??).

**Comment by kalessil** [comment]
Fixed
